File: yizhigou-service-interface/src/main/java/com/yizhigou/sellergoods/service/SpecificationService.java
package com.yizhigou.sellergoods.service;

import com.yizhigou.Extra.SpecificationExt;
import com.yizhigou.entity.PageResult;
import com.yizhigou.entity.Result;

import java.util.List;
import java.util.Map;

public interface SpecificationService {

    PageResult findPage(SpecificationExt specificationExt, int pageNum, int pageSize);

    Result add(SpecificationExt specificationExt);

    Result update(SpecificationExt specificationExt);

    SpecificationExt findOne(long id);

    void delete(Long[] ids);

    List<Map> selectSpecificationOptionList();
}

File: yizhigou-manager-web/src/main/webapp/js/service/specificationService.js
app.service("specificationService", function ($http) {

    this.search = function (page,rows,searchEntity) {
        return $http.post('../specification/search.do?page=' + page + '&rows=' + rows,searchEntity);
    }

    this.findOne = function (id) {
        return $http.post('../specification/findOne.do?id=' + id);
    }

    this.delete = function (ids) {
        return $http.post('../specification/delete.do?ids=' + ids);
    }

    this.add = function (entity) {
        return $http.post("../specification/add.do" , entity);
    }

    this.update = function (entity) {
        return $http.post("../specification/update.do" , entity);
    }

    this.selectSpecificationOptionList = function () {
        return $http.get("../typeTemplate/selectSpecificationOptionList.do");
    }
})